client/engine/messageservice/test-messageservice_test.go

```diff
@@ -0,0 +1,52 @@
+package messageservice
+
+import (
+	"testing"
+
+	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/protocols"
+	directfund "github.com/statechannels/go-nitro/protocols/direct-fund"
+	"github.com/statechannels/go-nitro/types"
+)
+
+var aliceMS = TestMessageService{
+	address: types.Address{'a'},
+	toPeers: make(map[types.Address]chan<- protocols.Message),
+
+	in:  make(chan protocols.Message),
+	out: make(chan protocols.Message),
+}
+
+var bobMS = TestMessageService{
+	address: types.Address{'b'},
+	toPeers: make(map[types.Address]chan<- protocols.Message),
+
+	in:  make(chan protocols.Message),
+	out: make(chan protocols.Message),
+}
+
+var objective, _ = directfund.New(state.TestState, aliceMS.address)
+var testId protocols.ObjectiveId = "testObjectiveID"
+
+var aToB protocols.Message = protocols.Message{
+	To:          bobMS.address,
+	ObjectiveId: testId,
+	Sigs:        make(map[*state.State]state.Signature),
+	Proposal:    objective,
+}
+
+func TestConnect(t *testing.T) {
+	aliceMS.Run()
+	bobMS.Run()
+	bobIn := bobMS.GetReceiveChan()
+
+	aliceMS.Connect(bobMS)
```

Comment:
In a fully connected test messaging network with N peers, I think we would need to call `Connect` N(N-1) times, creating N**2 - N `chans`. This is probably fine because `chans` are cheap. But I was wondering if you had considered having a more centralised solution -- i.e. each peer has a send and receive to a single message bus (2N `chans` in total). I think this is called pub/sub https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern.

Reply:
Idea being that per-peer connection more closely resembles real world networking system. We expect agents in a state channel network communicate p2p, so why not mirror that here?